### DISCUSSION TITLE
List private rooms in Space directory

### DISCUSSION THREAD

**Discussion by MarkWieczorek** [element-hq/element-meta#911] [Ideas]
#### What would you like to do?

When creating a room, you have an option of setting the visibility/access to one of these three options 

1. Private. Only invited people can join.
2. Space members. Anyone in a space can find and join.
3. Public. Anyone can find and join.

However, when I invite someone to a space, I would like them to see a list of all available rooms, even though some might be private and need an invitation to join.

#### Why would you like to do it?
When joining a space, you simply do not know which private rooms exist. If you knew that they existed, you could then ask the room admin to let you join.

#### How would you like to achieve it?
Create a new room access/visibility. One way to do this would be to divide Option 1 above into two.

1a. Private. Only invited people can find and join.
1b. Private but findable. Only invited people can join, but anyone can find (or anyone in the space can find).

alternatively, you could divide 2 into
2a. Space members. Anyone in a space can find and join.
2b Space members, private. Anyone in a space can find but only invited people can join.

**Comment by t3chguy** [comment]
This might be doable with the knock join rule, where a user can request to join a private room, those may be visible in the /hierarchy API
